connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -262,11 +269,23 @@ private static ConfigKeyInfo convertConfigKey(ConfigKey configKey) {
         String width = configKey.width.name();
         String displayName = configKey.displayName;
         List<String> dependents = configKey.dependents;
-        return new ConfigKeyInfo(name, type, required, defaultValue, importance, documentation, group, orderInGroup, width, displayName, dependents);
+        return new ConfigKeyInfo(name, typeName, required, defaultValue, importance, documentation, group, orderInGroup, width, displayName, dependents);
     }
 
-    private static ConfigValueInfo convertConfigValue(ConfigValue configValue) {
-        return new ConfigValueInfo(configValue.name(), configValue.value(), configValue.recommendedValues(), configValue.errorMessages(), configValue.visible());
+    private static ConfigValueInfo convertConfigValue(ConfigValue configValue, Type type) {
+        String value = ConfigDef.convertToString(configValue.value(), type);
+        List<String> recommendedValues = new LinkedList<>();
+
+        if (type == Type.LIST) {
```

Comment:
I'm confused what's going on here. Why would we special case lists here? Shouldn't recommendedValues for a LIST type be a list of lists, i.e. it is a list of valid list values? And in that case a normal `convertToString` call will work fine -- each list value will be passed through a `",".join(recommendedList)` and be converted to a string, and the overall list of recommended values will therefore be a list of strings when put into the JSON output.

Maybe this confusion also explains the special casing of LIST earlier?